apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentDetailsDialog.tsx

```diff
@@ -16,179 +16,375 @@ import {
+          ({ screeningQuestion }) =>
+            screeningQuestion.id === existingQuestion?.id,
+        );
+      })
+      .map<Scalars["ID"]>((q) => q.id);
+
+    // question in new list doesn't have an ID yet - create it
+    const toBeCreatedInputs = newScreeningQuestions
+      .filter(({ screeningQuestion }) => !screeningQuestion.id)
+      .map<CreateScreeningQuestionInput>(({ index, screeningQuestion }) => ({
+        question: {
+          en: screeningQuestion.en,
+          fr: screeningQuestion.fr,
+        },
+        sortOrder: index + 1,
+      }));
+
+    // question in new list already has an ID - submit for update
+    const toBeUpdatedInputs = newScreeningQuestions
+      .filter(({ screeningQuestion }) => notEmpty(screeningQuestion.id))
+      .map<UpdateScreeningQuestionInput>(({ index, screeningQuestion }) => ({
+        id: screeningQuestion.id ?? "",
+        question: {
+          en: screeningQuestion.en,
+          fr: screeningQuestion.fr,
+        },
+        sortOrder: index + 1,
+      }));
+
+    const mutationParameters = {
+      id: values.poolId,
+      pool: {
+        screeningQuestions: {
+          update: toBeUpdatedInputs,
+          create: toBeCreatedInputs,
+          delete: toBeDeletedIds,
+        },
+      },
+    };
+
+    return executeUpdatePoolMutation(mutationParameters).then((res) => {
+      if (res?.data?.updatePool?.id) {
+        return Promise.resolve();
+      }
+      return Promise.reject();
+    });
+  };
+
+  const submitForm = (values: FormValues) => {
+    const assessmentStepPromise =
+      dialogAction === "update"
+        ? submitUpdateAssessmentStepMutation(values)
+        : submitCreateAssessmentStepMutation(values);
+
+    const combinedPromise =
+      dialogMode === "screening_question"
+        ? assessmentStepPromise.then(() =>
+            submitUpdateScreeningQuestionMutation(values),
+          )
+        : assessmentStepPromise;
+
+    combinedPromise
+      .then(() => {
+        toast.success(
+          intl.formatMessage({
+            defaultMessage: "Successfully saved assessment step!",
+            id: "W1vWDi",
+            description:
+              "Success message displayed after unlinking an experience to a skill",
+          }),
+        );
+        setIsOpen(false);
+        reset(); // the create dialog could be used several times in a row
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Error: saving assessment step failed.",
+            id: "DnXch4",
+            description:
+              "Message displayed to user after assessment step fails to be saved.",
+          }),
+        );
+      });
   };
 
   const canAddScreeningQuestions =
     fields.length < SCREENING_QUESTIONS_MAX_QUESTIONS;
 
-  const assessedSkillsItems: CheckboxOption[] = allPoolSkills
-    .map((poolSkill) => ({
+  const assessedSkillsItems: CheckboxOption[] = allPoolSkills.map(
```

Comment:
Unsure of the best approach here but, if you have a pool in draft mode with no skills, it simply renders an empty skill selection check list which looks odd and never validates.

![Screenshot 2023-11-06 151432](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/dcb5e87a-de9a-4958-b0f1-fed807ac5ecf)

Reply:
We could probably use a null state here like:
`There are no skills selected. Please use the 'Advertisement information' page to add some skills.`

Reply:
Added the message in.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/6a31a0ee-d5ff-4cf2-a943-ee60bcef55bd)
[error for no skills](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8364/commits/01ad45f1ca812a97353f602c6518f341e741fb8c)